src/client/components/robot/rate.ts

```diff
@@ -0,0 +1,58 @@
+import { observable } from 'mobx'
+import { runInAction } from 'mobx'
+import { throttle } from 'throttle-debounce'
+
+import { Clock, getClock } from '../../../shared/time/clock'
+
+type RateOpts = {
+  smoothing: number
+  unitTime: number
+}
+
+export class Rate {
+  @observable rate = 0
+  private currentRate: number = 0
+  private smoothing: number
+  private unitTime: number
+  private currentUpdateCount: number = 0
+  private lastUpdateTime?: number
+
+  constructor({ smoothing, unitTime }: RateOpts, private clock: Clock) {
+    this.smoothing = smoothing
+    this.unitTime = unitTime
+  }
+
+  static of(opts: RateOpts) {
+    return new Rate(opts, getClock())
```

Comment:
This file is (and will always be) in `/client`, so it will always be a `BrowserSystemClock`. So I suggest we just avoid the indirection.

```suggestion
    return new Rate(opts, BrowserSystemClock)
```

Reply:
That's what I had initially. But that led to an issue I didn't know how to solve:

Since `Rate` is created indirectly when `RobotModel` is created, there's no opportunity to specify the type of clock to use.

But since it's a better idea to move the stats and rate stuff from `RobotModel`, we can use `BrowserSystemClock` here 👍.

Reply:
Updated in https://github.com/NUbots/NUsight2/pull/269/commits/d45eebc07e27f8ee6ec63ec12ad609812f2ae61b